test/Broadway/EventSourcing/EventSourcingRepositoryTest.php

```diff
@@ -78,6 +81,34 @@ public function it_throws_an_exception_if_the_static_method_does_not_exist()
         $repository->load('y0l0');
     }
 
+    /**
+     * @test
+     */
+    public function it_publishes_decorated_events()
+    {
+        $projector = new TestMetadataPublishedProjector();
+        $this->eventBus->subscribe($projector);
+
+        $repository = new EventSourcingRepository(
+            $this->eventStore,
+            $this->eventBus,
+            '\Broadway\EventSourcing\TestEventSourcedAggregate',
+            new PublicConstructorAggregateFactory(),
+            array(new MetadataEnrichingEventStreamDecorator(array(new TestDecorationMetadataEnricher())))
```

Comment:
Can you explain further? The repository needs a decorator and the decorator needs an array of enrichers. I'm not sure what you meant by your comment, nor why moving it would change how it's constructed.